### PR TITLE
bring back fields for editing user and group search bases 

### PR DESCRIPTION
**Pull request by jsuchome**
(bnc#815506)
- 2.24.1

### CODE REVIEW

File: src/Ldap.ycp [hunk 1]
```diff
@@ -2275,6 +2283,11 @@ Select another one.
 	SCR::Write (add (domain, "ldap_user_uuid"), "entryuuid");
 	SCR::Write (add (domain, "ldap_group_uuid"), "entryuuid");
 
+	SCR::Write (add (domain, "ldap_user_search_base"),
+	  (nss_base_passwd != base_dn && nss_base_passwd != "") ?  nss_base_passwd : nil);
```
Comment (jreidinger) [inline src/Ldap.ycp]: I don't get much what is reason to write nil if `nss_base_passwd == base_dn`. Maybe some comment can help here. I know that it is in help text, but I would like to see here reason

Comment (jsuchome) [inline src/Ldap.ycp]: Writing nil removes the line. If there's default value (= same as base_dn), it is better to remove the key.,

File: src/ui.ycp [hunk 2]
```diff
@@ -1039,19 +1083,22 @@ have changed your configuration.</p>
...
+	if (result == `client || result == `admin || result == `searching)
 	{
 	    current = (symbol) result;
 	    Wizard::SetHelpText (help_text[current]:"");
 	    if (result == `client)
 		set_client_term ();
-	    else
+	    else if (result == `admin)
 		set_admin_term ();
+	    else
```
Comment (jreidinger) [inline src/ui.ycp]: In general it is potentially problematic using default branch when you don't want default branch, but only searching branch. It add high coupling between upper if and result if. I also think that using switch can help here

```
switch(result)
{
  case `client:
    set_client_term ();
    break;
  case `admin:
    set_admin_term ();
    break;
  case `searching:
    set_searching_term ();
    break;
  default:
    y2internal("unknown event in tab switching");
}
```

File: src/ui.ycp [hunk 3]
```diff
@@ -1067,7 +1114,12 @@ have changed your configuration.</p>
 		]);
 	    if (dn != "")
 	    {
-		UI::ChangeWidget (`id(`base_config_dn), `Value, dn);
+		map br2entry	= $[
```
Comment (jreidinger) [inline src/ui.ycp]: again you have coupling in code between br2entry and upper contains call. ideally br2entry should be constant map and contains should look into its keys, so there is only one place which you need modify.

File: src/Ldap.ycp (hunk 1 above)
Comment (jreidinger) [inline src/Ldap.ycp]: I understand code. I just think that you should add comment, that base_dn is default value for pwd or group, so you should remove it. It is not obvious from first look

File: src/ui.ycp (hunk 2 above)
Comment (jsuchome) [inline src/ui.ycp]: I do not like our switch; I'll add if condition there. And you and see that the block already prevents any other value, so default branch is useless.

**Comment by jreidinger** [comment]
LGTM

**Pull request merged**